src/app/public/modules/tabs/tabset-adapter.service.ts

```diff
@@ -12,25 +12,42 @@ export class SkyTabsetAdapterService {
 
   private bntsEl: HTMLElement;
 
+  private tabsOffsetLeft: number;
+
   public init(elRef: ElementRef) {
-    this.el = elRef.nativeElement.querySelector('.sky-tabset');
-    this.tabsEl = elRef.nativeElement.querySelector('.sky-tabset-tabs');
-    this.bntsEl = elRef.nativeElement.querySelector('.sky-tabset-btns');
+    const nativeElement = elRef.nativeElement;
+    this.el = nativeElement.querySelector('.sky-tabset');
+    this.tabsEl = nativeElement.querySelector('.sky-tabset-tabs');
+    this.bntsEl = nativeElement.querySelector('.sky-tabset-btns');
+
+    this.tabsOffsetLeft = this.getTabsOffsetLeft();
 
     this.detectOverflow();
   }
 
-  public detectOverflow() {
+  public detectOverflow(): void {
     if (this.el && this.tabsEl) {
       let elWidth = this.el.offsetWidth;
-      let tabsElWidth = this.tabsEl.offsetWidth + this.bntsEl.offsetWidth;
+      let tabsElWidth = this.tabsEl.offsetWidth + this.bntsEl.offsetWidth + this.tabsOffsetLeft;
 
-      const currentOverflow = (tabsElWidth > elWidth);
+      const areTabsOverflowing = (tabsElWidth > elWidth);
 
-      if (this.currentOverflow !== currentOverflow) {
-        this.currentOverflow = currentOverflow;
-        this.overflowChange.emit(currentOverflow);
+      if (this.currentOverflow !== areTabsOverflowing) {
+        this.currentOverflow = areTabsOverflowing;
+        this.overflowChange.emit(this.currentOverflow);
       }
     }
   }
+
+  /**
+   * Returns the number of pixels to the left of the first tab.
+   */
+  private getTabsOffsetLeft(): number {
+    const tabsetRect = this.el.getBoundingClientRect();
+    const firstTabOrDropdownRect = this.el
+      .querySelector('span:first-child')
```

Comment:
Looking for a `span` seems a little brittle - especially if a dev slips another `span` in there at a later time. Is there a way we could better ensure we're grabbing the right thing? Could we looks for `sky-tab-dropdown` or `sky-tab-button` elements? If those elements don't have bounding rectangles, then maybe the specific classes are better? (`.sky-tabset-dropdown` / `sky-tabset-tabs`)?

Reply:
Good call. Actually, the dropdown always exists in the DOM, so I can reference that CSS class.